packages/playwright-core/src/utils/isomorphic/urlMatch.ts

```diff
@@ -101,7 +102,33 @@ export function urlMatches(baseURL: string | undefined, urlString: string, match
     // Allow http(s) baseURL to match ws(s) urls.
     if (baseURL && /^https?:\/\//.test(baseURL) && /^wss?:\/\//.test(urlString))
       baseURL = baseURL.replace(/^http/, 'ws');
-    match = constructURLBasedOnBaseURL(baseURL, match);
+
+    const tokenMap = new Map<string, string>();
+    function mapToken(original: string, replacement: string) {
+      tokenMap.set(replacement, original);
+      return replacement;
+    }
+    // Glob symbols may be escaped in the URL and some of them such as ? affect resolution,
+    // so we replace them with safe components first.
+    const relativePath = match.split('/').map((token, index) => {
+      if (token === '.' || token === '..' || token === '')
+        return token;
+      // Handle special case of http*://
+      if (index === 0 && token.endsWith(':'))
+        return mapToken(token, 'http:');
```

Comment:
```suggestion
        return mapToken(token, 'pw-matcher-schema:');
```

Reply:
Unfortunately, it didn't work. The slashes are properly inserted after domain only for known schemas (http:, ws:).

Reply:
It has to be a "special scheme", defined here: https://url.spec.whatwg.org/#special-scheme